### PR TITLE
Task/mt 7183 implement w3 w google map saved list pin

### PR DESCRIPTION
**Pull request by kaytran2992**

### CODE REVIEW

File: lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt [hunk 1]
```diff
@@ -232,43 +234,76 @@ class W3WMapManager(
...
-        zoomLevel: Float? = null
-    ) {
-        CoroutineScope(dispatcher).launch {
-            when (val c23wa = textDataSource.convertToCoordinates(words)) {
+    suspend fun addMarkerAtListWords(
+        listId: String,
+        listWords: List<String>,
+        listColor: W3WMarkerColor,
+        onSuccess: ((List<W3WAddress>) -> Unit)? = null,
+        onError: ((List<W3WError>) -> Unit)? = null,
```
Comment (phpduy) [inline lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:242]: We don't need these callbacks if the function is a suspending function

File: lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt [hunk 2]
```diff
@@ -232,43 +234,76 @@ class W3WMapManager(
...
-        listWords: List<String>,
+    suspend fun addMarkerAtWords(
+        words: String,
         markerColor: W3WMarkerColor = MAKER_COLOR_DEFAULT,
         zoomOption: W3WZoomOption = W3WZoomOption.CENTER_AND_ZOOM,
-        onSuccess: Consumer<List<W3WAddress>>? = null,
-        onError: Consumer<W3WError>? = null
-    ) {
+        onSuccess: ((W3WAddress) -> Unit)? = null,
+        onError: ((W3WError) -> Unit)? = null,
```
Comment (phpduy) [inline lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt:284]: We don't need these callbacks if the function is a suspending function

File: lib-compose/src/main/java/com/what3words/components/compose/maps/models/W3WMarker.kt [hunk 3]
```diff
@@ -9,7 +9,21 @@ data class W3WMarker(
     val color: W3WMarkerColor,
     val title: String? = null,
     val snippet: String? = null
-)
+) {
+    val id: Long = generateUniqueId()
```
Comment (phpduy) [inline lib-compose/src/main/java/com/what3words/components/compose/maps/models/W3WMarker.kt:13]: We can make this property lazy by using lazy {}

File: lib-compose/src/main/java/com/what3words/components/compose/maps/providers/googlemap/W3WGoogleMapDrawer.kt [hunk 4]
```diff
@@ -47,6 +45,11 @@ fun W3WGoogleMapDrawer(
     mapConfig: W3WMapDefaults.MapConfig,
     onMarkerClicked: (W3WMarker) -> Unit
 ) {
+    // Check if selectedMarker exists in the saved list
+    val markerStatus = remember(state.listMakers, state.selectedAddress) {
```
Comment (phpduy) [inline lib-compose/src/main/java/com/what3words/components/compose/maps/providers/googlemap/W3WGoogleMapDrawer.kt:49]: Can we use derivedStateOf here?

**Review by phpduy — CHANGES_REQUESTED** [review]

**Review by phpduy — APPROVED** [review]
LGTM!

**Review by QuocNguyen-w3w — APPROVED** [review]
LGTM

**Pull request merged**